test/unit/math/mix/fun/eigenvalues_test.cpp

```diff
@@ -15,5 +15,19 @@ TEST(mathMixFun, eigenvalues) {
   EXPECT_THROW(f(a32), std::invalid_argument);
 }
 
+TEST(mathMixFun, eigenvaluesComplex) {
+  auto f = [](const auto& x) {
+    using stan::math::eigenvalues;
+    return eigenvalues(stan::math::to_complex(x, 0));
+  };
+  for (const auto& x : stan::test::square_test_matrices(0, 2)) {
+    stan::test::expect_ad(f, x);
+  }
+
+  Eigen::MatrixXd a32(3, 2);
+  a32 << 3, -5, 7, -7.2, 9.1, -6.3;
+  EXPECT_THROW(f(a32), std::invalid_argument);
+}
+
```

Comment:
Can you add a test for complex number that isn't a real number? Just to make sure this is returning values that you expect.